packages/circuits/test/bigComparators.test.ts

```diff
@@ -127,31 +128,45 @@ describe('Compare large numbers', function () {
         }
     })
 
-    it('should compare equal upper bits', async () => {
-        const n = random() >> BigInt(64)
-        const { isValid, publicSignals } = await genProofAndVerify(
-            'upperComparators' as any,
-            {
-                in: [n.toString(), n.toString()],
-            }
-        )
-        expect(isValid).to.be.true
-        expect(publicSignals[0].toString()).to.equal('0')
+    it('should compare equal lower bits', async () => {
+        const n = randomf(BigInt(2) ** BigInt(64))
+        const n1 = n + (randomf(BigInt(2) ** BigInt(150)) << BigInt(64))
+        const n2 = n + (randomf(BigInt(2) ** BigInt(150)) << BigInt(64))
```

Comment:
can this be not hardcoding the magic number `150` and `64`?

Reply:
like here
even if we change the epoch to `48` bits
https://github.com/Unirep/Unirep/blob/4bd82e81e2082b0d57f7339a858398e0e006f191/packages/circuits/circuits/epochKeyLite.circom#L52
It is hard to detect the bits are mismatched

Reply:
Sure, i applied a small fix here [5173d38](https://github.com/Unirep/Unirep/pull/449/commits/5173d3870d8d31274491ac6fe6eee701a377af3c)